src/main/java/org/dcsa/tnt/service/impl/TNTEventServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package org.dcsa.tnt.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.dcsa.core.events.model.EquipmentEvent;
+import org.dcsa.core.events.model.Event;
+import org.dcsa.core.events.model.ShipmentEvent;
+import org.dcsa.core.events.model.TransportEvent;
+import org.dcsa.core.events.repository.EventRepository;
+import org.dcsa.core.events.repository.PendingEventRepository;
+import org.dcsa.core.events.service.EquipmentEventService;
+import org.dcsa.core.events.service.GenericEventService;
+import org.dcsa.core.events.service.ShipmentEventService;
+import org.dcsa.core.events.service.TransportEventService;
+import org.dcsa.core.events.service.impl.GenericEventServiceImpl;
+import org.dcsa.core.exception.NotFoundException;
+import org.dcsa.core.extendedrequest.ExtendedRequest;
+import org.dcsa.core.service.impl.ExtendedBaseServiceImpl;
+import org.dcsa.tnt.service.TNTEventService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Service
+public class TNTEventServiceImpl extends GenericEventServiceImpl implements TNTEventService {
+
+    @Autowired
+    private PendingEventRepository pendingEventRepository;
+    @Autowired
+    private EventRepository eventRepository;
+
```

Comment:
@douradofrazer do we need both:
private PendingEventRepository pendingEventRepository;
and
private EventRepository eventRepository;
?
I thought we only needed one of them???

Reply:
You're right we don't need `private EventRepository eventRepository;` since it's being set in `GenericEventServiceImpl`

Reply:
if it is being set in GenericEventServiceImpl - do we then need getModelClass() and getRepository()? Myt guess is that they are also set in GenericEventSerivceImpl?

Reply:
Form what I understand `GenericEventServiceImpl` extends `ExtendedBaseServiceImpl`
which then implements `ExtendedBaseService` which extends `BaseService` which has an implementation for `getModel()` so we might not need it IMO. Does it cause issues if not specified @kagemand-andersen ?
`GenericEventServiceImpl` only overrides `getRepository()`

Reply:
@douradofrazer I have removed it and it made no difference. It's an override that is identical to that of the `GenericEventServiceImpl` so it make sense that it would just use that one.